packages/lodestar/src/chain/validation/attestation.ts

```diff
@@ -1,177 +1,286 @@
-    throw new AttestationError({
-      code: AttestationErrorCode.INVALID_INDEXED_ATTESTATION,
-      job: attestationJob,
-    });
+  // [REJECT] The signature of attestation is valid.
+  const indexedAttestation: phase0.IndexedAttestation = {
+    attestingIndices: [validatorIndex] as List<number>,
+    data: attData,
+    signature: attestation.signature,
+  };
+  const signatureSet = getIndexedAttestationSignatureSet(attestationTargetState, indexedAttestation);
+  if (!(await chain.bls.verifySignatureSets([signatureSet]))) {
+    throw new AttestationError({code: AttestationErrorCode.INVALID_SIGNATURE});
   }
 
-  if (!ssz.Epoch.equals(attestation.data.target.epoch, computeEpochAtSlot(attestationSlot))) {
-    throw new AttestationError({
-      code: AttestationErrorCode.BAD_TARGET_EPOCH,
-      job: attestationJob,
-    });
+  // Now that the attestation has been fully verified, store that we have received a valid attestation from this validator.
+  //
+  // It's important to double check that the attestation still hasn't been observed, since
+  // there can be a race-condition if we receive two attestations at the same time and
+  // process them in different threads.
+  if (chain.seenAttesters.isKnown(targetEpoch, validatorIndex)) {
+    throw new AttestationError({code: AttestationErrorCode.ATTESTATION_ALREADY_KNOWN, targetEpoch, validatorIndex});
   }
 
-  try {
-    if (!isCommitteeIndexWithinRange(attestationTargetState, attestation.data)) {
-      throw new AttestationError({
-        code: AttestationErrorCode.COMMITTEE_INDEX_OUT_OF_RANGE,
-        index: attestation.data.index,
-        job: attestationJob,
-      });
-    }
-  } catch (error) {
-    throw new AttestationError({
-      code: AttestationErrorCode.COMMITTEE_INDEX_OUT_OF_RANGE,
-      index: attestation.data.index,
-      job: attestationJob,
-    });
-  }
+  chain.seenAttesters.add(targetEpoch, validatorIndex);
 
-  if (!doAggregationBitsMatchCommitteeSize(attestationTargetState, attestation)) {
-    throw new AttestationError({
-      code: AttestationErrorCode.WRONG_NUMBER_OF_AGGREGATION_BITS,
-      job: attestationJob,
-    });
-  }
+  return {indexedAttestation, subnet: expectedSubnet};
+}
 
-  if (!chain.forkChoice.isDescendant(attestation.data.target.root, attestation.data.beaconBlockRoot)) {
-    throw new AttestationError({
-      code: AttestationErrorCode.TARGET_BLOCK_NOT_AN_ANCESTOR_OF_LMD_BLOCK,
-      job: attestationJob,
-    });
+/**
+ * Verify that the `attestation` is within the acceptable gossip propagation range, with reference
+ * to the current slot of the `chain`.
+ *
+ * Accounts for `MAXIMUM_GOSSIP_CLOCK_DISPARITY`.
+ */
+export function verifyPropagationSlotRange(chain: IBeaconChain, attestationSlot: Slot): void {
+  // TODO (LH): Do not queue future attestations for later processing. Investigate why LH doesn't do it
+  // TODO: use MAXIMUM_GOSSIP_CLOCK_DISPARITY to compute the clock.currentSlot
```

Comment:
remove these 2 TODOs as we count on `MAXIMUM_GOSSIP_CLOCK_DISPARITY_SEC` already